### PR TITLE
chore(l10n): combine the download l10n scripts

### PR DESCRIPTION
**Pull request by chenba**
Because:
 - we have multiple similar scripts that pull translations from the
   fxa-content-server-l10n repo
 - we need that functionality for new settings

This commit:
 - combine the scripts into a single one

Part of #7319

### CODE REVIEW

**Comment by chenba** [comment]
An earlier version of this patch tried to check the l10n repo into a shared location.  That was a bit of a disaster since yarn execute things in parallel, so there were overlapping calls to the script.  This version doesn't change the location of the checkout for any module, thus very low risk.

**Comment by dannycoates** [comment]
> That was a bit of a disaster since yarn execute things in parallel, so there were overlapping calls to the script.

Not that I think we should change this PR, but would putting the script in the top-level package.json postinstall work?

**Comment by chenba** [comment]
> 
> would putting the script in the top-level package.json postinstall work?

I think so.  At least I cannot think of an obvious reason why it wouldn't.

File: _scripts/clone-l10n.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+fi
+
+if [ -n "$FXA_L10N_SKIP" ]; then
+    echo "Skipping fxa-content-server-l10n update..."
+    exit 0
+fi
+
+DOWNLOAD_PATH="https://github.com/mozilla/fxa-content-server-l10n.git"
+SCRIPT_DIR=$(dirname "$0")
+ROOT_DIR=$(realpath "$SCRIPT_DIR/..")
```
Comment (dannycoates) [inline _scripts/clone-l10n.sh:19]: `realpath` doesn't exist on mac. I think we should be ok using paths relative to `SCRIPT_DIR`

File: _scripts/clone-l10n.sh [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+  esac
+
+  for src in locale/**/$FTL_FILENAME.ftl; do
+    [ -f "$src" ] || continue
+    dir=$(dirname "$src")
+    dir=$(echo "$dir" | sed "s/_/-/")
+    dir=$(basename "$dir")
+    base=$(basename "$src")
+    mkdir -p "$MODULE_PATH/public/locales/$dir"
+    cp "$src" "$MODULE_PATH/public/locales/$dir/$base"
```
Comment (dannycoates) [inline _scripts/clone-l10n.sh:61]: If `$MODULE_PATH` is relative, this can be `"../public/locales/$dir"`

**Review by dannycoates — APPROVED** [review]
pre-approving but needs a fix for mac. it works on mac after i tweaked the paths in the script to be relative. i'd like to get to a single dir for fxa-content-server-l10n someday

**Comment by chenba** [comment]
> 
> pre-approving but needs a fix for mac. 

Thanks.  Not writing shell script often enough to keep this compatibility stuff in mind.  :-/

> i'd like to get to a single dir for fxa-content-server-l10n someday

If you mean simply having a single checkout dir, then that should be pretty easy.  But, we'd still need to copy stuff around.  auth-server's mail translator requires the translations directly, so in order to keep the relative path from the translator to the translations constant, we'd need to copy fxa-content-server-l10n.  (Of course we could change translator itself...)  

refs: 
 - https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/lib/senders/translator.js#L26
 - https://github.com/mozilla/fxa/blob/main/packages/fxa-auth-server/package.json#L12

**Pull request merged**